chargeOclock/Source/Other/NetworkAPI.swift

```diff
@@ -47,14 +57,24 @@ class NetworkAPI {
 		performRequest(to: uri, method: "DELETE", completion: completion)
 	}
 
-	private func performRequest(to uri: String, method httpMethod: String, completion: @escaping NetworkCompletion) {
+	private func performRequest(to uri: String, method httpMethod: String, header: [String : String]? = nil, body : [String : String]? = nil, completion: @escaping NetworkCompletion) {
```

Comment:
if you leave it as is, that body is passed in as an optional data structure (fix the type, though), the headers parameter isn't needed anyway, since we know we need to add the JSON mimetype header anyway and can do that here if body is passed in.
although, I'd rename body to json (in all methods)